ruby.c

```diff
@@ -829,12 +841,44 @@ process_sflag(int sflag)
             if (s[1] == '-' && s[2] == '\0')
                 break;
 
+            // Handle `-abc` in extended switch mode. Also, removes the leading `-` for long-form
+            // `--foo`s in extended mode so they end up as `$foo`, not `$_foo`.
+            if (gflag && *++s != '-') {
+                while (*s != '\0') {
+                    char name = *s++;
+                    if (name != '_' && !ISALPHA(name)) {
+                        VALUE name_error[2];
+                        name_error[0] = rb_str_new2("invalid name for global variable - ");
+                        char error_flag[3] = { '-', name, '\0' };
```

Comment:
This can make incomplete multibyte string.

Reply:
What's a good way to check for this? I could just do the remainder of the string, but that'd also bring in any other characters